### PR TITLE
Sort segments in process detection result #354

### LINKED ISSUE #354: Sort segments in processDetectionResult
In `analytic_controller.ts` in `processDetectionResult` method we don't sort segments, but use `_.last()` that choose wrong segment for webhook

### PR DESCRIPTION
**Pull request by amper43**
* sort segments in `processDetectionResult`
* send only non-sended last segment

fixes #354

### CODE REVIEW

File: server/src/controllers/analytics_controller.ts [hunk 1]
```diff
@@ -305,9 +305,19 @@ async function processDetectionResult(analyticUnitId: AnalyticUnit.AnalyticUnitI
       `Missing lastDetectionTime in result or it is corrupted: ${JSON.stringify(detectionResult)}`
     );
   }
-  console.debug(`got detection result with ${detectionResult.segments.length} segments`);
+  console.debug(`got detection result for ${analyticUnitId} with ${detectionResult.segments.length} segments`);
 
-  const segments = detectionResult.segments.map(
+  let sortedSegments: {from, to}[] = detectionResult.segments;
+  sortedSegments = sortedSegments.sort((a, b) => {
```
[the remark this answers is not in the document]
Comment (amper43) [reply server/src/controllers/analytics_controller.ts]: yes, it would be bettr, thanks

File: server/src/services/alert_service.ts [hunk 2]
```diff
@@ -13,7 +13,17 @@ export class Alert {
   public update(now: number) {};
 }
 
-class PatternAlert extends Alert {};
+class PatternAlert extends Alert {
+
+  private lastSendedSegment: Segment;
```
Comment (rozetko) [inline server/src/services/alert_service.ts]: `lastSentSegment`?:)

File: server/src/services/alert_service.ts [hunk 3]
```diff
@@ -13,7 +13,17 @@ export class Alert {
...
+
+  public recieve(segment: Segment) {
+    if(this.lastSendedSegment === undefined || !segment.equals(this.lastSendedSegment) ) {
+      this.lastSendedSegment = segment;
+      this.sender(this.analyticUnit, segment);
```
Comment (rozetko) [inline server/src/services/alert_service.ts]: `sender` doesn't seem like a good name for a function
function name should be a verb (`send`)

File: server/src/services/alert_service.ts [hunk 4]
```diff
@@ -13,7 +13,17 @@ export class Alert {
   public update(now: number) {};
 }
 
-class PatternAlert extends Alert {};
+class PatternAlert extends Alert {
+
+  private lastSendedSegment: Segment;
+
+  public recieve(segment: Segment) {
```
Comment (rozetko) [inline server/src/services/alert_service.ts]: `receive`
¯\_(ツ)_/¯

**Pull request merged**